File: app/src/main/java/app/devlife/connect2sql/db/model/SqlModel.java
package app.devlife.connect2sql.db.model;

import android.content.ContentValues;
import android.database.Cursor;

import net.sqlcipher.database.SQLiteDatabase;

public interface SqlModel<T> {

    Class<T> getModelClass();

    String getTableName();

    String getCreateSql();

    T hydrateObject(Cursor cursor) throws IllegalAccessException, InstantiationException;

    ContentValues toContentValues(T object);
}

File: app/src/main/java/app/devlife/connect2sql/db/model/SqlModel.kt
package app.devlife.connect2sql.db.model

import android.content.ContentValues
import android.database.Cursor

import net.sqlcipher.database.SQLiteDatabase

interface SqlModel<T> {

    val modelClass: Class<T>

    val tableName: String

    val createSql: String

    fun upgradeSql(oldVersion: Int, newVersion: Int): List<String>

    @Throws(IllegalAccessException::class, InstantiationException::class)
    fun hydrateObject(cursor: Cursor): T

    fun toContentValues(`object`: T): ContentValues
}
